packages/core/src/chart.ts

```diff
@@ -171,7 +171,7 @@ export class Chart {
 					this.model,
 					this.services,
 					[
-						legendComponent,
+						...((this.model.getOptions().legend.visible !== false) ? [ legendComponent ] : [ ]),
```

Comment:
I'd prefer if you stored` this.model.getOptions().legend.visible` and added a condititional after line 149 so that the checks for layout directions are not performed (since they are unneeded if the chart is the only thing in the fullFrameComponent).

the conditional would then bypass the codeblock below if there isnt going to be a legend

```
const legendPosition = Tools.getProperty(this.model.getOptions(), "legend", "position");
		if (legendPosition === "left") {
			fullFrameComponentDirection = LayoutDirection.ROW;

			if (!this.model.getOptions().legend.orientation) {
				this.model.getOptions().legend.orientation = LegendOrientations.VERTICAL;
			}
		} else if (legendPosition === "right") {
			fullFrameComponentDirection = LayoutDirection.ROW_REVERSE;

			if (!this.model.getOptions().legend.orientation) {
				this.model.getOptions().legend.orientation = LegendOrientations.VERTICAL;
			}
		} else if (legendPosition === "bottom") {
			fullFrameComponentDirection = LayoutDirection.COLUMN_REVERSE;
		}
```

Reply:
ok this should be fixed now